### PR TITLE
Custom Objections

### PR DESCRIPTION
**Pull request by cents02**
Added multiple custom objections + /clear which clears ooc.
**How to use**
Right click on the custom objection button and available options should appear
**How to enable**
Rename "custom" to "custom1" the .gif and wav files. For additional ones name them "custom2","custom3" etc.
**Planned updates**
You can name those custom objections. I will do that whenever I return back to my  main workstation.

**Important note**
This update should be followed along with my **PR in tsuserver3** in order to work

### CODE REVIEW

File: src/courtroom.cpp [hunk 1]
```diff
@@ -897,9 +897,12 @@ void Courtroom::enter_courtroom(int p_cid)
...
 
   if (ao_app->custom_objection_enabled &&
-      (file_exists(ao_app->get_character_path(current_char, "custom1.gif")) ||
+      ((file_exists(ao_app->get_character_path(current_char, "custom1.gif")) ||
       file_exists(ao_app->get_character_path(current_char, "custom1.apng"))) &&
-      file_exists(ao_app->get_character_path(current_char, "custom1.wav")))
+      file_exists(ao_app->get_character_path(current_char, "custom1.wav"))) ||
+      ((file_exists(ao_app->get_character_path(current_char, "custom.gif")) ||
+      file_exists(ao_app->get_character_path(current_char, "custom.apng"))) &&
+      file_exists(ao_app->get_character_path(current_char, "custom.wav"))))
```
Comment (oldmud0) [inline src/courtroom.cpp]: This logic is overloaded.

Comment (cents02) [reply src/courtroom.cpp]: I know, this why I tried to not have the custom and custom1 as options for this reason but you told me to add backwards comp. I will probs end up scrapping this method anyway.

**Comment by cents02** [comment]
@oldmud0 I wanted to use 41(4 -> custom objection indication 1-> first C.O + I have commented wherever it's used in order to be obvious what it does) as I didn't see where else I could fit it in the packet. The other solution/method I see is to either:

- Extend the packet struct to do something like this: #4#custom obj name# (This could also enable us to add a feature which a char can do two emotes one after the other)

Or

-Keep the enumeration of 41,42.. but have the custom objections also assigned in the char.ini so C.O1  -> gotcha.gif, CO2 -> silence etc

**Comment by oldmud0** [comment]
I'd go with the first one, but since we can't add any more parameters horizontally using `#`, we can use the alternate `&` delimiter instead.

Proposed protocol change:

```diff
 ### objection_modifier - 10

 Dictates if the player uses an objection, hold it, take that or a custom shout

 - 0 = nothing
 - 1 = hold it
 - 2 = objection
 - 3 = take that
-- 4 = custom shout (AO2 only)
+- 4 = custom shout (AO2 only) (format: `4&shout_name`)
+  - If no shout name is specified, then `custom` is used by default.

**Comment by cents02** [comment]
Sounds good, and I assume I make another category in the char.ini to grab the custom objection names?

**Comment by stonedDiscord** [comment]
how about
40 for custom.gif
41 for custom1.gif

**Comment by cents02** [comment]
@stonedDiscord even if I set it to 40 longbyte issue is that the name of the custom objection should be shown rather having them custom1,custom2 etc

**Pull request closed without merging**